cli/internal/fs/turbo_json.go

```diff
@@ -432,17 +433,21 @@ func (btd *BookkeepingTaskDefinition) UnmarshalJSON(data []byte) error {
 	btd.TaskDefinition.TopologicalDependencies = []string{} // TODO @mehulkar: this should be a set
 	btd.TaskDefinition.TaskDependencies = []string{}        // TODO @mehulkar: this should be a set
 
+	// If there was a dependsOn field, add the bookkeeping
+	// we don't care what's in the field, just that it was there
+	// We'll use this marker to overwrite while merging TaskDefinitions.
+	if task.DependsOn != nil {
+		btd.definedFields.Add("DependsOn")
+	}
```

Comment:
validated that this fixes the issue in vercel/front I was seeing. Test in that repo:

```bash
$ ~/dev/vercel/turbo/target/debug/turbo run unit-test --skip-infer --filter=playwright --dry=json | jq '.tasks | map(select(.taskId=="playwright#unit-test")) | .[0].resolvedTaskDefinition'
```

result after:

```diff
{
  "outputs": [],
  "cache": true,
-  "dependsOn": ["^build"],
+  "dependsOn": [],
  "inputs": ["**/*.{js,jsx,ts,tsx}"],
  "outputMode": "full",
  "env": [],
  "persistent": false
}
```